### PR TITLE
Mirth recency - persist datetime for recency last successful submission date

### PR DESCRIPTION
**Pull request by echiteri**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java [hunk 1]
```diff
@@ -76,8 +78,16 @@ public void execute() {
...
-			
-			if (strSubmissionDate.equals(dateFormat.format(todayDate)) && strSubmitOnceDaily.equals("true")) {
+			Date gpSubmissionDate = null;
+			try {
+				gpSubmissionDate = new SimpleDateFormat("yyyy-MM-dd").parse(strSubmissionDate);
+			}
+			catch (ParseException e) {
+				e.printStackTrace();
+			}
+			System.out.println("submission date: " + gpSubmissionDate);
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java]: @echiteri use a logger here

File: api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java [hunk 2]
```diff
@@ -57,6 +58,7 @@
 	public void execute() {
 		Date todayDate = new Date();
 		DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+		DateFormat dateTimeFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java]: @echiteri rename the date and time format variables to make sense - what are you using them for?

Comment (echiteri) [reply api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java]: @ssmusoke isn't dateFormat just denote the format for date and dateTimeFormat the format for date and time?

File: api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java [hunk 3]
```diff
@@ -76,8 +78,16 @@ public void execute() {
...
 		
 		if (!isBlank(strSubmissionDate)) {
-			
-			if (strSubmissionDate.equals(dateFormat.format(todayDate)) && strSubmitOnceDaily.equals("true")) {
+			Date gpSubmissionDate = null;
+			try {
+				gpSubmissionDate = new SimpleDateFormat("yyyy-MM-dd").parse(strSubmissionDate);
+			}
+			catch (ParseException e) {
+				e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (echiteri) [reply api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java]: done

File: api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java (hunk 1 above)
Comment (echiteri) [reply api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java]: done

File: api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java (hunk 2 above)
Comment (ssmusoke) [reply api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java]: Okay then

File: api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (ssmusoke) [reply api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java]: Also add the stack trace, and clarify which date the parsing has failed and also add the value that failed to parse

File: api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java (hunk 1 above)
Comment (ssmusoke) [reply api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java]: Remove the System.out.printlin

File: api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (echiteri) [reply api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java]: done

File: api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java (hunk 1 above)
Comment (ssmusoke) [reply api/src/main/java/org/openmrs/module/ugandaemrsync/tasks/SendRecencyDataToCentralServerTask.java]: @echiteri this is still open

**Comment by slubwama** [comment]
@echiteri  toooooooo many commits. Close this PR and open up an new one with a branch created from upstream.

**Pull request merged**